DatadogInternal/Sources/Models/SessionReplay/PrivacyLevel+SessionReplay.swift

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import UIKit
+
+// MARK: - DatadogExtension for UIView
+
+/// Extension on `DatadogExtension` to provide access to `SessionReplayOverride` for any `UIView`.
+extension DatadogExtension where ExtendedType: UIView {
+    /// Provides access to Session Replay override settings for the view.
+    /// Usage: `myView.dd.sessionReplayOverride.textAndInputPrivacy = .maskNone`.
+    public var sessionReplayOverride: SessionReplayOverrideExtension<ExtendedType> {
+        get {
+            return SessionReplayOverrideExtension(self.type)
+        }
+        set {}
+    }
+}
+
+// MARK: - Associated Keys
+
+private var associatedTextAndInputPrivacyKey: UInt8 = 3
+private var associatedImagePrivacyKey: UInt8 = 4
+private var associatedTouchPrivacyKey: UInt8 = 5
+private var associatedHiddenPrivacyKey: UInt8 = 6
+
+// MARK: - SessionReplayOverrideExtension
+
+/// `UIView` extension  to manage the Session Replay privacy override settings.
+public struct SessionReplayOverrideExtension<ExtendedType> {
```

Comment:
@ganeshnj made a similar comment. I placed the extension in `DatadogInternal` because that's where the privacy levels are defined. Do you know what was the reasoning for placing them in the internal module initially?

Reply:
Thanks for the suggestion! But the constraint on `UIView` is wanted, as we specifically want these properties to be available only on `UIView`. Don't you think?

Reply:
Ys, the constraint for `UIView` stays:
```swift
extension DatadogExtension where ExtendedType: UIView 
```
What I'm proposing is to make `SessionReplayOverrideExtension` a class and to use `where ExtendedType: AnyObject` so you don't have to do `view as AnyObject` in associated object.
To go even further, you don't really need a generic type here. You can directly use `private let view: UIView`, unless I'm missing something.

Reply:
Ok, I get it now. It makes sense, I will use your suggestion.